homeassistant/components/rest/binary_sensor.py

```diff
@@ -114,16 +124,72 @@ def is_on(self):
         """Return true if the binary sensor is on."""
         return self._is_on
 
+    @property
+    def extra_state_attributes(self):
+        """Return the state attributes."""
+        return self._attr_extra_state_attributes
+
     def _update_from_rest_data(self):
         """Update state from the rest data."""
-        if self.rest.data is None:
+        response = self.rest.data
+        _LOGGER.debug("Data fetched from resource: %s", response)
+
+        if response is None:
             self._is_on = False
 
-        response = self.rest.data
+        if self.rest.headers is not None:
+            # If the http request failed, headers will be None
```

Comment:
Should the state of the binary sensor be `unknown` under this circumstance?

Reply:
You mean in case `response` is `None`?

I would have thought so too, but this was the original behavior and changing it could present a breaking change for some users. The logic behind it might be that `bool(None)` gives `False`. So no response is an implicit `False` response.

Reply:
well considering the `const.py` approach, you can also revert back to use the full type...